glslc/src/file_compiler.cc

```diff
@@ -168,8 +170,30 @@ bool FileCompiler::EmitCompiledResult(
     }
   }
 
+  // Should write the result as a C-style initializer list if:
+  //  1) the result is SPIRV binary code;
+  //  2) the compiler is set to emit result as a C-style initializer list;
+  //  3) the compiler is not dumping dependency info as compilation output.
+  const bool should_emit_spirv_binary_as_c_initializer_list =
+      std::is_same<CompilationResultType,
+                   shaderc::SpvCompilationResult>::value &&
+      emit_binary_as_c_init_list_ &&
+      !(dependency_info_dumping_handler_ &&
+        dependency_info_dumping_handler_->DumpingAsCompilationOutput());
+
   // Write compilation output to output file.
-  out->write(compilation_output.data(), compilation_output.size());
+  if (should_emit_spirv_binary_as_c_initializer_list) {
+    // The output should be written to the output stream as a C-style
+    // initializer list.
+    *out << "{";
+    for (uint32_t word : result) {
+      *out << "0x" << std::uppercase << std::setfill('0') << std::setw(8)
+           << std::hex << word << ",";
```

Comment:
I'd prefer to wrap after every 4 words.
That will make it easier to inspect the result.
Also, I think we shouldn't emit the trailing comma.